stan-dev/rstan
# Merge accumulated CRAN requests from RStan 2.21.x

#### Summary:

Over the months and years, CRAN requested several changes for RStan 2.21.x that I had to implement and then upload to CRAN. Where applicable, we need to continue those going forward.

#### Intended Effect:

The main changes were:

- Don't utilize tbbmalloc_proxy . We were only using this on Macs because it made things noticeably faster, but CRAN didn't like it and the RStudio maintainers of RcppParallel stated that they did not really intend for tbbmalloc_proxy to be used for CRAN packages, as opposed to something that an individual might enable locally.
- The V8 javascript library caused a lot of friction, so I disabled it for 2.21.x. We now have a PR to use QuickJSR, so I'll merge that next.
- Copying the Stan (algorithms) library into the RStan sources. This is probably a good idea, although I reverted it. No package on CRAN seems to be using the copy in StanHeaders so moving it would be painless.
- Make `extract_sparse_parts` compatible with recent releases of the Matrix package. This seems to have gone smoothly, although we got a duplicate PR about it that I will close once this goes through.

#### How to Verify:

These have all passed `R CMD check` on CRAN, although there are some merge conflicts now that I will have to fix before the unit tests will run again.

#### Side Effects:

Slower on Mac but nothing else should be noticeable.

#### Documentation:

There were no substantive changes to the documentation, although there were a couple non-substantive ones.

#### Reviewer Suggestions: 

This does not really need to be reviewed either, since all of it is already fine on CRAN.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Trustees of Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: GPLv3 (https://opensource.org/licenses/GPL-3.0)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
